tools/base/checker.py

```diff
@@ -0,0 +1,285 @@
+import argparse
+import os
+from functools import cached_property
```

Comment:
The other high-level comment is that maybe a README.md would make sense explaining the structure of the tooling and what is "internal" (i.e. most people don't care about) and what is going to be used by devs.

Reply:
yep, agreed, ill add a README today with some examples on how to create a  `MyRunner` and a `MyChecker`

Reply:
just realising that i already added a README with some info about the `Runner` [here](https://github.com/envoyproxy/envoy/tree/main/tools#using-the-toolsbaserunnerrunner-class)

im also noticing some errors in that doc - ill update it add some info about the `Checker` class

Reply:
readme updated